### PR TITLE
Add point group information to FCIDUMP interface

### PR DESCRIPTION
**Pull request by fevangelista**
## Description
This PR perfects the `fcidump` interface by adding a field that stores the point group symmetry (`PNTGRP`, just to keep it in the FORTRAN spirit of "Thou shall use no unnecessary vowels in variable names"). For example:
```
ORBSYM=1,1,1,1,4,4,4,3,3,3,2,2,2,
ISYM=1,
PNTGRP=D2H,
```
Why is this change necessary? In essence because with the current information it is impossible to restart a computation that uses symmetry. Without the point group information, one cannot determine the psi4 irrep labels of orbitals from the standard information contained in the FCIDUMP file (see issue #1431) because when the file is written the irreps are resorted in a way that depends on the point group.

One potential issue: codes that consume FCIDUMP files produced by psi4 could encounter problems parsing the line that starts with `PNTGRP`. So I made this extra line optional via an argument passed to the `fcidump()` python function

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Add point group information to fcidump files (enabled via bool argument)
- [x] Fix a bug in the function that reads fcidump files when no orbital energies are present.

## Checklist
- [x] Updated `fcidump` test

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by hokru** [comment]
Tests fail because of an unrelated error (https://github.com/psi4/psi4/issues/1943).

File: psi4/driver/p4util/fcidump.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 from .exceptions import ValidationError, TestComparisonError
 
 
-def fcidump(wfn, fname='INTDUMP', oe_ints=None):
+def fcidump(wfn, fname='INTDUMP', oe_ints=None, write_pntgrp=False):
     """Save integrals to file in FCIDUMP format as defined in Comp. Phys. Commun. 54 75 (1989)
     Additional one-electron integrals, including orbital energies, can also be saved.
```
Comment (robertodr) [inline psi4/driver/p4util/fcidump.py:44]: Could you add to the docstring that the `PNTGRP` label is not in the Comp Phys Chem standard and might be incompatible with other codes?

**Comment by fevangelista** [comment]
I just found a few things to improve so let's put this on pause for a few days.

**Comment by fevangelista** [comment]
I have messed up this repo, so I created a new pull request #1951.

**Pull request closed without merging**